### DISCUSSION TITLE
Is there a way to have rx.markdown render mermaid diagrams?

### DISCUSSION THREAD

**Discussion by hragbalian** [reflex-dev/reflex#3125] [Q&A]
Hi there, 

I would like to be able to add mermaid diagrams to my markdown but the rx.markdown doesn't render it.  

For example, the following code:

```
```mermaid
    graph TD
        A[Start] -->|Get books| B[Go to library]
        B --> C{Books available?}
        C -->|Yes| D[Borrow books]
        C -->|No| E[Reserve books]
        D --> F[Read books]
        E --> F
        F --> G[Return books]
        G --> H[Finish]
```

should yield:

```mermaid
    graph TD
        A[Start] -->|Get books| B[Go to library]
        B --> C{Books available?}
        C -->|Yes| D[Borrow books]
        C -->|No| E[Reserve books]
        D --> F[Read books]
        E --> F
        F --> G[Return books]
        G --> H[Finish]
```

but instead it's rendering as 

<img width="972" alt="Screenshot 2024-04-20 at 10 29 13 AM" src="https://github.com/reflex-dev/reflex/assets/5995255/d83ce61b-6c97-4cec-aa7f-5db1f0fb1835">

Thanks!

**Comment by picklelo** [comment]
Looks like this is possible if we add a remark plugin: https://github.com/remarkjs/react-markdown/issues/394

You can see here where we add the plugins: https://github.com/reflex-dev/reflex/blob/19007ee2a101cab3cdee4e55d4f198811e80c7f2/reflex/components/markdown/markdown.py#L31

We could add this plugin also, or figure out a way to make the plugins dynamic so the user could set the ones they want themselves. I'll convert this to an issue in case someone wants to take this on.

**Reply by datavistics** [reply]
That would be a cool feature!

**Comment by clemlesne** [comment]
See: https://github.com/reflex-dev/reflex/issues/3137.
